libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -36,11 +38,13 @@ namespace solidity
 
 const unsigned CompilerUtils::dataStartOffset = 4;
 const size_t CompilerUtils::freeMemoryPointer = 64;
+const size_t CompilerUtils::zeroPointer = CompilerUtils::freeMemoryPointer + 0x20;
 const unsigned CompilerUtils::identityContractAddress = 4;
 
 void CompilerUtils::initialiseFreeMemoryPointer()
 {
-	m_context << u256(freeMemoryPointer + 32);
+	solAssert(zeroPointer == CompilerUtils::freeMemoryPointer + 0x20, "");
+	m_context << u256(zeroPointer + 0x20);
```

Comment:
Can we have another constant for `userMemoryPointer` (perhaps better naming needed)?

Reply:
I don't really understand what you mean, perhaps we should have a quick call about this.